pyfr/backends/cuda/blasext.py

```diff
@@ -56,29 +56,43 @@ def errest(self, x, y, z, *, norm):
         if x.traits != y.traits != z.traits:
             raise ValueError('Incompatible matrix types')
 
-        # Wrap
-        xarr = GPUArray(x.leaddim*x.nrow, x.dtype, gpudata=x)
-        yarr = GPUArray(y.leaddim*y.nrow, y.dtype, gpudata=y)
-        zarr = GPUArray(z.leaddim*z.nrow, z.dtype, gpudata=z)
+        nrow, ldim, dtype = x.traits
+        ncola, ncolb = x.ioshape[1:]
 
-        # Norm type
-        reduce_expr = 'a + b' if norm == 'l2' else 'max(a, b)'
+        # Determine the block size
+        block = (128, 1, 1)
+
+        # Determine the grid size
+        grid = (int((ncolb + block[0] - 1)/block[0]), 1, 1)
```

Comment:
What is wrong with the get grid for block function?  Is a block size of 128 always going to work?